tests/test_async_s3_bucket.py

```diff
@@ -0,0 +1,119 @@
+# This file is part of ts_salobj.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import io
+import os
+
+import asynctest
+
+import boto3
+import moto
+
+from lsst.ts import salobj
+
+
+class AsyncS3BucketTest(asynctest.TestCase):
+    def setUp(self):
+        # Use moto.mock_s3 in "raw mode" because:
+        # * @moto.mock_s3() does not work on async test methods.
+        # * Even if it did, or if I used moto.mock_s3() as a context manager,
+        #   I'd have to make the s3 bucket inside the test code.
+        self.mock = moto.mock_s3()
+        self.mock.start()
+
+        # Set s3 authentication environment variables to bogus values
+        # to avoid any danger of writing to a real s3 server.
+        for key in ("AWS_ACCESS_KEY_ID",
+                    "AWS_SECRET_ACCESS_KEY",
+                    "AWS_SECURITY_TOKEN",
+                    "AWS_SESSION_TOKEN"):
+            os.environ[key] = "testing"
```

Comment:
Is it worth saving and resetting these as part of `tearDown`? Or does `os.environ` only live as long as this python instance?

Reply:
Changing os.environ only affects the Python subprocess.